packages/core/src/Drawer/Drawer.tsx

```diff
@@ -72,21 +78,27 @@ export const Drawer: React.FC<DrawerProps> = ({
               bottom: snapPosition,
               width: snapDimensions.width,
               height: snapDimensions.height,
+              zIndex: zIndex,
             }
           : {}
       }
       transition={{ type: 'spring', bounce: 0 }}
       drag='y'
       dragConstraints={{ top: 0, bottom: 0 }}
       onDragEnd={handleSnap}
+      {...(snapHeights ? { onDragEnd: handleSnap } : {})}
       data-testid='snap-container'
     >
       <AnimatePresence>
         {isOpen && (
           <DrawerWrapper
             placement={isMobile ? 'bottom' : placement}
             padding={isFloating ? 3 : 0}
-            maxHeight={isMobile ? ['290px', '400px', '480px', 'calc(100vh - 64px)'] : props.height ?? '100%'}
+            maxHeight={
+              isMobile && !disableMaxHeight
```

Comment:
On 2nd thoughts, wonder if instead of adding `disableMaxHeight` as a prop, we should add `maxHeight` as a prop instead

Reply:
That would make sense since it gives consumers more flexibility.